From de57785b84ccf84cc3fbe1dff4044ccf137b3fbd Mon Sep 17 00:00:00 2001
From: Ajay Kaher <ajay.kaher@broadcom.com>
Date: Wed, 26 Jun 2024 15:54:15 +0530
Subject: [PATCH] Add RPMTAG_VCS to optionalTags[]

macros provided to rpmbuild as arguments (i.e. with -D option) has been added
to rpm as tag if that is in the list of optionalTags[].

Adding RPMTAG_VCS to optionalTags[] so that it can be passed to rpmbuild
as argument (i.e. -D "phvcs VALUE") and gets included in rpm as RPMTAG_VCS tag.

Following is the format to pass phvcs as argument to rpmbuild command:  
-D "phvcs https://github.com/vmware/photon/commit/73ce593;https://github.com/vmware/photon/commit/4eb3351",

Following is the command to get the value of RPMTAG_VCS from rpm:
# rpm -q '--qf=%{vcs}' stage/RPMS/noarch/photon-release-5.0-310.ph5.noarch.rpm
https://github.com/vmware/photon/commit/73ce593;https://github.com/vmware/photon/commit/4eb3351

If no phvcs provided to rpmbuild command then RPMTAG_VCS will not be added to rpm and '(none)'
will be returned on query for VCS:
# rpm -q '--qf=%{vcs}' stage/RPMS/noarch/photon-release-5.0-310.ph5.noarch.rpm
(none)

---
 build/parsePreamble.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build/parsePreamble.c b/build/parsePreamble.c
index f3e415a..52bc71a 100644
--- a/build/parsePreamble.c
+++ b/build/parsePreamble.c
@@ -538,6 +538,7 @@ static struct optionalTag {
     { RPMTAG_TRANSLATIONURL,	"%{translationurl}" },
     { RPMTAG_UPSTREAMRELEASES,	"%{upstreamreleases}" },
     { RPMTAG_MODULARITYLABEL,	"%{modularitylabel}"},
+    { RPMTAG_VCS,		"%{phvcs}"},
     { -1, NULL }
 };
 
-- 
2.7.4

